osu.Game/Online/Chat/ChannelManager.cs

```diff
@@ -445,6 +445,15 @@ protected override Task Poll()
             return tcs.Task;
         }
 
+        public void MarkChannelAsRead(Message message)
+        {
+            var channel = JoinedChannels.First(c => c.Id == message.ChannelId);
```

Comment:
Probably better written as
```suggestion
            var channel = JoinedChannels.Single(c => c.Id == message.ChannelId);
```
so that it fails loudly if there are ever duplicate channel ids.

Reply:
I didn't know this LINQ method existed, I will try to use them in the future.